toolchain/check/merge.h

```diff
@@ -96,11 +96,20 @@ struct DeclParams {
 // Checks that the parameters in a redeclaration of an entity match the
 // parameters in the prior declaration. If not, produces a diagnostic if
 // `diagnose` is true, and returns false.
-auto CheckRedeclParamsMatch(
-    Context& context, const DeclParams& new_entity,
-    const DeclParams& prev_entity,
-    SemIR::SpecificId prev_specific_id = SemIR::SpecificId::None,
-    bool check_syntax = true, bool diagnose = true) -> bool;
+auto CheckRedeclParamsMatch(Context& context, const DeclParams& new_entity,
+                            const DeclParams& prev_entity,
+                            SemIR::SpecificId prev_specific_id,
+                            bool check_syntax, bool check_self, bool diagnose)
+    -> bool;
+
+inline auto CheckRedeclParamsMatch(Context& context,
+                                   const DeclParams& new_entity,
+                                   const DeclParams& prev_entity) -> bool {
+  return CheckRedeclParamsMatch(context, new_entity, prev_entity,
+                                SemIR::SpecificId::None, /*diagnose=*/true,
+                                /*check_syntax=*/true,
+                                /*check_self=*/true);
```

Comment:
```suggestion
                                SemIR::SpecificId::None, /*check_syntax=*/true,
                                /*check_self=*/true, /*diagnose=*/true);
```

Note because of the `/*` clang-format won't wrap parameters when they could be. But this is an ordering comment, which clang-tidy might also catch. :)